pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/NormalizedDateSegmentNameGenerator.java

```diff
@@ -53,6 +53,8 @@ public NormalizedDateSegmentNameGenerator(String tableName, @Nullable String seg
       boolean excludeSequenceId, @Nullable String pushType, @Nullable String pushFrequency,
       @Nullable DateTimeFormatSpec dateTimeFormatSpec) {
     _segmentNamePrefix = segmentNamePrefix != null ? segmentNamePrefix.trim() : tableName;
+    Preconditions.checkArgument(
+        _segmentNamePrefix == null || isValidSegmentName(_segmentNamePrefix));
```

Comment:
```suggestion
        _segmentNamePrefix != null && isValidSegmentName(_segmentNamePrefix));
```

Reply:
good catch. forgot to change after merging the 2 rules.